### PR TITLE
add kk980 keyboard

### PR DESCRIPTION
**Pull request by Sets3n**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sets3n/kk980/config.h [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* key combination for command */
+//#define IS_COMMAND() ( keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)))
+
+/* prevent stuck modifiers */
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/sets3n/kk980/config.h:70]:
```suggestion
```

File: keyboards/sets3n/kk980/config.h [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+#define RGB_DI_PIN B7
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/sets3n/kk980/config.h:74]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/sets3n/kk980/rules.mk [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# MCU name
+MCU = atmega32u4
+
+F_CPU = 16000000
```
Comment (drashna) [inline keyboards/sets3n/kk980/rules.mk:4]:
```suggestion
```

File: keyboards/sets3n/kk980/rules.mk [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+MOUSEKEY_ENABLE = yes	# Mouse keys(+4700)
+EXTRAKEY_ENABLE = yes	# Audio control and System control(+450)
+CONSOLE_ENABLE = no	# Console for debug(+400)
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = yes
+BLUETOOTH_ENABLE = no
```
Comment (drashna) [inline keyboards/sets3n/kk980/rules.mk:24]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes           # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
AUDIO_ENABLE = no
RGBLIGHT_ENABLE = yes
n```

**Review by drashna — CHANGES_REQUESTED** [review]
Missing a readme file.

File: keyboards/sets3n/kk980/config.h (hunk 2 above)
Comment (Sets3n) [reply keyboards/sets3n/kk980/config.h:68]: What's the meaning? Can I use the "RGBLIGHT_ANIMATIONS" option anymore?

Comment (drashna) [reply keyboards/sets3n/kk980/config.h:68]: Well, it was never meant to be used that way, actually.  Also, if more animations are added, it doesn't risk increasing the firmware size past the limit, at least for AVR and small capacity STM32 boards.

Comment (Sets3n) [reply keyboards/sets3n/kk980/config.h]: Oh I see.

File: keyboards/sets3n/kk980/readme.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# kk980
+
+![kk980](https://imgur.com/a/9JJjC77)
+
+*A short description of the keyboard/project*
+
+* Keyboard Maintainer: [sets3n](https://github.com/sets3n)
+* Hardware Supported: *The PCBs, controllers supported*
+* Hardware Availability: *Links to where you can find this hardware*
```
Comment (drashna) [inline keyboards/sets3n/kk980/readme.md:9]: These should be filled out.

Also, could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sets3n/kk980/readme.md (hunk 5 above)
Comment (Sets3n) [reply keyboards/sets3n/kk980/readme.md]: Thank you, I learned a lot.

Comment (Sets3n) [reply keyboards/sets3n/kk980/readme.md]: This is very helpful for me.

**Review by drashna — APPROVED** [review]

File: keyboards/sets3n/kk980/rules.mk [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+BOOTMAGIC_ENABLE = yes	# Enable Bootmagic Lite 
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no		# Console for debug
+COMMAND_ENABLE = no    	# Commands for debug and configuration
+SLEEP_LED_ENABLE = no  	# Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no  	# Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/sets3n/kk980/rules.mk:18]:
```suggestion
# Bootloader selection
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/sets3n/kk980/readme.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
+# kk980
+
+![kk980](https://imgur.com/a/9JJjC77)
```
Comment (fauxpark) [inline keyboards/sets3n/kk980/readme.md:3]: Please use a direct image link, and fix the orientation.

File: keyboards/sets3n/kk980/readme.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+Make example for this keyboard (after setting up your build environment):
+
+    `make kk980:default`
+    `make kk980:via`
+
+## FLASH
+
+Flashing example for this keyboard:
+
+    `make kk980:default:flash`
```
Comment (fauxpark) [inline keyboards/sets3n/kk980/readme.md:30]:
```suggestion
Make example for this keyboard (after setting up your build environment):

    make sets3n/kk980:default

Flashing example for this keyboard:

    make sets3n/kk980:default:flash
```

File: keyboards/sets3n/kk980/kk980.c [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+#include "kk980.h"
+
+void eeconfig_init_kb(void) {  // EEPROM is getting reset!
+    rgblight_enable();                      // Enable RGB by default
+    rgblight_mode(RGBLIGHT_MODE_RGB_TEST);  // set to RGBLIGHT_MODE_RGB_TEST by default
+
+    eeconfig_update_kb(0);
+    eeconfig_init_user();
+}
+
```
Comment (fauxpark) [inline keyboards/sets3n/kk980/kk980.c:26]: Can be replaced with defines: https://docs.qmk.fm/#/feature_rgblight?id=configuration

File: keyboards/sets3n/kk980/info.json [hunk 10]
```diff
@@ -0,0 +1,118 @@
+{
+    "keyboard_name": "kk980",
+    "url": "",
+    "maintainer": "Sets3n",
+    "width": 17,
+    "height": 6,
```
Comment (fauxpark) [inline keyboards/sets3n/kk980/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/sets3n/kk980/rules.mk [hunk 11]
```diff
@@ -0,0 +1,20 @@
+# MCU name
+MCU = atmega32u4
+# Bootloader selection
```
Comment (fauxpark) [inline keyboards/sets3n/kk980/rules.mk:3]:
```suggestion

# Bootloader selection
```

File: keyboards/sets3n/kk980/readme.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    `make sets3n/kk980:default`
+
+## FLASH
+
+Flashing example for this keyboard:
+
+    `make sets3n/kk980:default:flash`
```
Comment (fauxpark) [inline keyboards/sets3n/kk980/readme.md:29]:
```suggestion
    make sets3n/kk980:default

## FLASH

Flashing example for this keyboard:

    make sets3n/kk980:default:flash
```

File: keyboards/sets3n/kk980/readme.md [hunk 13]
```diff
@@ -0,0 +1,31 @@
+# kk980
+
+![kk980](https://i.imgur.com/MQA6FXb.jpg)
+
+A hotswap 98% RGB Mechanical keyboard.
+
+* Keyboard Maintainer: [sets3n](https://github.com/sets3n)
+* Hardware Supported: *kk980 PCB*
```
Comment (fauxpark) [inline keyboards/sets3n/kk980/readme.md:8]:
```suggestion
* Hardware Supported: kk980 PCB
```

File: keyboards/sets3n/kk980/readme.md [hunk 14]
```diff
@@ -0,0 +1,31 @@
+# kk980
+
+![kk980](https://i.imgur.com/MQA6FXb.jpg)
```
Comment (fauxpark) [inline keyboards/sets3n/kk980/readme.md:3]: This image is sideways.

File: keyboards/sets3n/kk980/kk980.c [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+#include "kk980.h"
+// TEST RGB FUNC
+// void eeconfig_init_kb(void) {  // EEPROM is getting reset!
+//     rgblight_enable();                      // Enable RGB by default
+//     rgblight_mode(RGBLIGHT_MODE_RGB_TEST);  // set to RGBLIGHT_MODE_RGB_TEST by default
+// 
+//     eeconfig_update_kb(0);
+//     eeconfig_init_user();
+// }
+
```
Comment (fauxpark) [inline keyboards/sets3n/kk980/kk980.c:26]:
```suggestion
```

File: keyboards/sets3n/kk980/rules.mk [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no		# Console for debug
+COMMAND_ENABLE = no    	# Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no  	# Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes		# USB Nkey Rollover
+BACKLIGHT_ENABLE = no  	# Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes	# Enable keyboard RGB underglow
+AUDIO_ENABLE = no		# Audio output
```
Comment (fauxpark) [inline keyboards/sets3n/kk980/rules.mk:20]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite 
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/sets3n/kk980/rules.mk [hunk 17]
```diff
@@ -0,0 +1,20 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#     change yes to no to disable
```
Comment (fauxpark) [inline keyboards/sets3n/kk980/rules.mk:8]:
```suggestion
#     change yes to no to disable
#
```

**Review by fauxpark — APPROVED** [review]

File: keyboards/sets3n/kk980/rules.mk [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
+
```
Comment (fauxpark) [inline keyboards/sets3n/kk980/rules.mk:9]:
```suggestion
#
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/sets3n/kk980/rules.mk (hunk 18 above)
Comment (Sets3n) [reply keyboards/sets3n/kk980/rules.mk]: Thank you. It made me a lot better

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**